src/pointer-analysis/goto_program_dereference.cpp

```diff
@@ -187,9 +187,7 @@ void goto_program_dereferencet::dereference_rec(
 
       exprt tmp;
       tmp.swap(expr.op0().op0());
-
-      if(tmp.type()!=expr.type())
-        tmp.make_typecast(expr.type());
+      tmp = typecast_exprt::conditional_cast(tmp, expr.type());
 
       expr.swap(tmp);
```

Comment:
I think that's `expr = typecast_exprt::conditional_cast(expr.op0().op0(), expr.type());`

Reply:
sure